cocotb/share/makefiles/Makefile.inc

```diff
@@ -133,6 +133,8 @@ manually override the LIBPYTHON_LOC make variable with the absolute path to libp
 
 endef
 
+ifndef LIBPYTHON_LOC
```

Comment:
Can one do?
```
LIBPYTHON_LOC ?= $(strip $(subst $(FIND_LIBPYTHON_RC)QQQQ,,$(FIND_LIBPYTHON_RES)QQQQ))
```

On the other hand, you do not want to call `cocotb-config --libpython` if not needed. 
Maybe a comment about this would be nice.

Something else:
If one rises execution (above) in `cocotb-config` maybe do not need all this makefile part/massage just?

`export LIBPYTHON_LOC ?= $(shell cocotb-config --libpython)`

Formating is not so nice but visible and makefile is cleaner but is a matter of taste.

Reply:
> `LIBPYTHON_LOC ?= $(strip $(subst $(FIND_LIBPYTHON_RC)QQQQ,,$(FIND_LIBPYTHON_RES)QQQQ))`

Well if `cocotb-config --libpython` fails the makefiles will still die here, even if the user specified a value for `LIBPYTHON_LOC`.

> `export LIBPYTHON_LOC ?= $(shell cocotb-config --libpython)`

The issue with this is that failing shell calls don't cause make to stop. It will: print the error message, `LIBPYTHON_LOC` will be empty (I think exceptions are printed on stderr), execution will continue. I would still need to check the return code in this instance to prevent execution from continuing.

We need to:
1. Allow the user override `LIBPYTHON_LOC`
2. Have the test fail iff `cocotb-config --libpython` fails and the user didn't set `LIBPYTHON_LOC`
3. (Optional) Avoid calling `cocotb-config` if `LIBPYTHON_LOC` is specified.

The current solution is the one that hits all the points with the smallest diff. I'm also backporting these commits onto 1.5.

Reply:
👍🏻